api/block/schema.py

```diff
@@ -1,16 +1,43 @@
 import graphene
-
+from graphql import GraphQLError
 from graphene_sqlalchemy import SQLAlchemyObjectType
 from api.block.models import Block as BlockModel
 from api.room.schema import Room
+from api.office.models import Office
 from helpers.room_filter.room_filter import room_join_location
+from utilities.utility import validate_empty_fields
+from helpers.auth.authentication import Auth
 
 
 class Block(SQLAlchemyObjectType):
     class Meta:
         model = BlockModel
 
 
+class CreateBlock(graphene.Mutation):
+    class Arguments:
+        name = graphene.String(required=True)
+        office_id = graphene.Int(required=True)
+    block = graphene.Field(Block)
+
+    @Auth.user_roles('Admin')
+    def mutate(self, info, **kwargs):
+        validate_empty_fields(**kwargs)
+        block_name = BlockModel.query.filter_by(name=kwargs['name'].title()).all() # noqa
+        if len(block_name) > 0:
```

Comment:
In stead of checking the length of the list after querying the blocks model with the name, you can simply use truthy test, something like if if block_name(this is called implicit boleanness and is more pythonic code)